### PR TITLE
scripts: add checkenv.sh for server environment checking

### PR DESCRIPTION
**Pull request by andelf**
- A ``checkenv.sh`` script.
- Update README

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+- OS, kernel version
+- CPU cores, HT status
+- Memory
+- Swap turned off
+- Disk space, mount options, rotational (may not be recognized correctly)
+- sysctl.conf
+- ulmit
+
+### How to use
```
Comment (shenli) [inline scripts/README.md]: Does this scripts work on all OS?

File: scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+```
+cd /path/to/tidb-tools/scripts
+./checkenv.sh
```
Comment (shenli) [inline scripts/README.md]: Do we need to run chmod?

File: scripts/README.md (hunk 1 above)
Comment (andelf) [inline scripts/README.md]: - Darwin / Win32 => prints "Happy Hacking"

**Comment by qiuyesuifeng** [comment]
How about make a bin dir, then all binaries and scripts can be in one path?

File: scripts/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+Kernel checking ... Linux ok
+Archetecture ... x86_64 ok
+Kernel version ... 3.19.0-69-generic ok
+CPU checking ... ok
+  Logical CPU Number  : 8
+  Physical CPU Number : 1
+  CPU Core Number     : 4
+  HT Number           : 2
+Total Memory  ... 15953 MiB ok
+Available Memory  ... skip
```
Comment (qiuyesuifeng) [inline scripts/README.md]: Maybe this check can be removed.

File: scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+Swap off ... ok
+Checking disks ...
+  sda1 of size  228_GB is mounted on /
```
Comment (qiuyesuifeng) [inline scripts/README.md]: _ is a little strange.

File: scripts/checkenv.sh [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+
+check_kernel_arch () {
+    local _uname_m=$(uname -m)
+    echo -n "${_uname_m} "
+    case "${_uname_m}" in
+	x86_64)
+	    echo_info ok
+	    ;;
+	i686)
+	    echo_warning works, for test only
```
Comment (qiuyesuifeng) [inline scripts/checkenv.sh]: Seems we have not tested this. @shenli

File: scripts/checkenv.sh [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+	    echo_warning "happy hacking!"
+	    ;;
+    esac
+}
+
+check_kernel_version () {
+    local _uname_r=$(uname -r)
+    echo -n "${_uname_r} "
+    case "${_uname_r}" in
+	2.6.*|3.*|4.*)
```
Comment (qiuyesuifeng) [inline scripts/checkenv.sh]: I think we can refer from docker.

File: scripts/checkenv.sh [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+## Memory
+# >= 8G
+check_memory_total () {
+    local _memory_total=$(cat /proc/meminfo | grep -i MemTotal | awk '{print $2}')
+    echo -n "$((${_memory_total} / 1024)) MiB "
+    if [ "${_memory_total}" -lt $((1024 * 1024 * 8)) ]; then
+	echo_warning "not sufficient"
+    else
```
Comment (qiuyesuifeng) [inline scripts/checkenv.sh]: If the value is very small, should we exit?

File: scripts/checkenv.sh [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+	echo_info ok
+    fi
+}
+
+# make sure swap is off
+check_swap () {
+    local _swap=$(cat /proc/meminfo | grep -i SwapTotal | awk '{print $2}')
+    local _swappiness=$(cat /proc/sys/vm/swappiness)
+    if [ "${_swap}" -ne 0 ]; then
+        echo
```
Comment (qiuyesuifeng) [inline scripts/checkenv.sh]: `echo` can be removed?

**Comment by andelf** [comment]
If we use Ansible for binary deployment. 
This PR will be deprecated by [Fact](http://docs.ansible.com/ansible/playbooks_variables.html#id21).

File: scripts/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+- Disk space, mount options, rotational (may not be recognized correctly)
+- sysctl.conf
+- ulmit
+
+NOTE: requires ``bash`` to run.
+
+### How to use
+
+```
+cd /path/to/tidb-tools/scripts
```
Comment (andelf) [inline scripts/README.md]: change to relative path

File: scripts/README.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+> ./checkenv.sh
+Kernel checking ... Linux ok
+Archetecture ... x86_64 ok
+Kernel version ... 3.19.0-69-generic ok
+CPU checking ... ok
+  Logical CPU Number  : 8
+  Physical CPU Number : 1
+  CPU Core Number     : 4
+  HT Number           : 2
+Total Memory  ... 15953 MiB ok
```
Comment (andelf) [inline scripts/README.md]: fixed

File: scripts/README.md (hunk 3 above)
Comment (andelf) [inline scripts/README.md]: fixed

File: scripts/README.md (hunk 4 above)
Comment (andelf) [inline scripts/README.md]: removed and use [228GB]

File: scripts/checkenv.sh (hunk 7 above)
Comment (andelf) [inline scripts/checkenv.sh]: fixed

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by iamxy** [comment]
LGTM

**Pull request merged**